dynamic-ssl/src/test/java/org/wildfly/security/dynamic/ssl/DynamicSSLTestUtils.java

```diff
@@ -0,0 +1,214 @@
+            keyStore.load(new FileInputStream(keystorePath), password.toCharArray());
+
+            // Create key manager
+            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KEY_MANAGER_FACTORY_ALGORITHM);
+            keyManagerFactory.init(keyStore, password.toCharArray());
+            KeyManager[] km = keyManagerFactory.getKeyManagers();
+
+            KeyStore trustStore = KeyStore.getInstance(KEYSTORE_TYPE);
+            trustStore.load(new FileInputStream(truststorePath), password.toCharArray());
+            // Create trust manager
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(KEY_MANAGER_FACTORY_ALGORITHM);
+            trustManagerFactory.init(trustStore);
+            TrustManager[] tm = trustManagerFactory.getTrustManagers();
+
+            // Initialize SSLContext
+            SSLContext sslContext = SSLContext.getInstance(TLS_PROTOCOL_VERSION);
+            sslContext.init(km, tm, null);
+
+            return sslContext;
+        } catch (Exception ex) {
+            Assert.fail();
+        }
+        return null;
+    }
+
+    static void createKeystores() throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
+        if (!KEYSTORES_DIR.exists()) {
+            KEYSTORES_DIR.mkdirs();
+        }
+        generateTwoWaySSLKeystoresAndTruststores(CLIENT1_KEYSTORE_FILENAME, SERVER1_KEYSTORE_FILENAME, CLIENT1_TRUSTSTORE_FILENAME, SERVER1_TRUSTSTORE_FILENAME);
+        generateTwoWaySSLKeystoresAndTruststores(CLIENT2_KEYSTORE_FILENAME, SERVER2_KEYSTORE_FILENAME, CLIENT2_TRUSTSTORE_FILENAME, SERVER2_TRUSTSTORE_FILENAME);
+        generateTwoWaySSLKeystoresAndTruststores(CLIENT3_KEYSTORE_FILENAME, SERVER3_KEYSTORE_FILENAME, CLIENT3_TRUSTSTORE_FILENAME, SERVER3_TRUSTSTORE_FILENAME);
+        generateTwoWaySSLKeystoresAndTruststores(DEFAULT_CLIENT_KEYSTORE_FILENAME, DEFAULT_SERVER_KEYSTORE_FILENAME, DEFAULT_CLIENT_TRUSTSTORE_FILENAME, DEFAULT_SERVER_TRUSTSTORE_FILENAME);
+    }
+
+    private static void generateTwoWaySSLKeystoresAndTruststores(String clientKeystoreFilename, String serverKeystoreFilename,
+                                                                 String clientTruststoreFilename, String serverTruststoreFilename) throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
+        // Generates client certificate and keystore
+        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+        KeyStore clientKeyStore = KeyStore.getInstance(KEYSTORE_TYPE);
+        clientKeyStore.load(null, null);
+
+        KeyPair clientKeyPair = keyPairGenerator.generateKeyPair();
+        PrivateKey signingKey = clientKeyPair.getPrivate();
+        PublicKey publicKey = clientKeyPair.getPublic();
+
+        X500Principal testClient10DN = new X500Principal("CN=" + CLIENT_ALIAS);
+        X509Certificate clientCertificate = new X509CertificateBuilder()
```

Comment:
Just wondering, could we make use of the `CAGenerationTool` to generate the certificates required for this test? (We make use of this in other TLS related tests to simplify things a bit.)

Reply:
Note that this doesn't block getting this merged though.

Reply:
@fjuma I did not check properly but I don't see why not, I will create a subsequent jira to address this once this gets merged. Thanks!

Reply:
or I will create it now and will put `depend on` this issue, thanks!

Reply:
created: https://issues.redhat.com/browse/ELY-2738

Reply:
Perfect, thanks!